frontend/src/components/ArticleForm.js

```diff
@@ -12,13 +12,16 @@ const ArticleForm = () => {
     e.preventDefault();
 
     const article = { title, summary, body, author: "alex" };
-    const response = await fetch("/api/articles/create", {
-      method: "POST",
-      body: JSON.stringify(article),
-      headers: {
-        "Content-Type": "application/json",
-      },
-    });
+    const response = await fetch(
+      `${process.env.REACT_APP_API_URL}/api/articles/create`,
```

Comment:
No! This is not the website URL but the API url, that the frontend tries to contact to create an article in this case.
Two things here:
- In development mode, `REACT_APP_API_URL` is set to an empty string. We will then fetch `/api/articles/create` here. This will work because of [this line](https://github.com/atxr/iscsc.fr/blob/v0.2.0/production-mode/frontend/package.json#L2). It creates a proxy between the frontend and the backend. Thanks to the proxy, the frontend can contact the backend as a local url.
- In production mode, the proxy doesn't work (I think this is because we build the frontend (~ "compile" the react code with webpack into vanilla js) so we don't have access to these fancy development features anymore). Then, we need to create a `reverse proxy` to make it work. To do so, I use `nginx`, which is deployed [here](https://github.com/atxr/iscsc.fr/blob/v0.2.0/production-mode/frontend/Dockerfile#L13)

I must confess it's quite new for me and I still have many things I didn't understand. So your questions are also helping me :)
I'll also talk about CORS here when you'll be ok with this first point

Reply:
Ow god that's much more complicated that I thought in the first place :sweat_smile: 

I got the development mode explanation but I must confess I don't understand **why** we need to make it work differently in production mode? why does react needs to be "compiled"?
And how does a `reverse proxy` is different from the development mode `proxy`? 
And would you have some `nginx` documentation for me?  

Yeah I know that's a lot of question :sweat_smile: 

> I must confess it's quite new for me and I still have many things I didn't understand. So your questions are also helping me :)

Learning is the point of all that so that's cool!

> I'll also talk about CORS here when you'll be ok with this first point

Should we go in a discussion channel to develop these point further?

Reply:
> I got the development mode explanation but I must confess I don't understand why we need to make it work differently in production mode? why does react needs to be "compiled"?

That's a good point, and the answer is because of performance issues. Running your app in dev mode for production is very bad regarding performances, because it's optimized to have good logs, provide good errors... and not to be blazingly fast and light. It does matter a lot for folks with slow internet; the lighter the better for them. Regarding performances, it's awful to have a slow website, that's not something you want :vomiting_face: 

> And how does a reverse proxy is different from the development mode proxy?

Before today, I was also confused about this, but now I have a better understanding I think. But explaining this will take a lot of time with a single message like this :thinking:
I will explain it during the web introduction next week if you want, or in a call!
But TLDR: it's the same thing, but you can do much more with nginx. And I actually need much more because of docker networking
About nginx documentation, you should just read (the nginx config file I wrote)[https://github.com/atxr/iscsc.fr/blob/v0.2.0/production-mode/nginx/nginx.conf.template] and think about nginx like a connection dealer: it listens on some ports and deals the connection to the appropriate app based on some rules (port, url...)

> Should we go in the discussion panel to develop these point further?
It can be a good idea to talk about nginx, docker and CORS indeed!

Reply:
> That's a good point, and the answer is because of performance issues. Running your app in dev mode for production is very bad regarding performances, because it's optimized to have good logs, provide good errors... and not to be blazingly fast and light. It does matter a lot for folks with slow internet; the lighter the better for them. Regarding performances, it's awful to have a slow website, that's not something you want vomiting_face

That seems logic after all :smile: 

Ok so let's [go](https://github.com/iScsc/iscsc.fr/discussions/23)!